crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -315,11 +473,9 @@ impl NativePriceEstimating for CachingNativePriceEstimator {
             }
 
             self.0
-                .estimate_prices_and_update_cache(&[token], self.0.max_age, 1)
-                .next()
-                .await
-                .unwrap()
-                .1
+                .blocking_estimate_prices_and_update_cache(&token, self.0.max_age)
```

Comment:
Here was the evil part. This was doing a direct call to the estimator. Now it does a blocking operation.